test/factory/common_factory_tests.dart

```diff
@@ -209,12 +210,12 @@ void _childKeyWarningTests(Function factory) {
 
       expect(consoleErrorCalled, isTrue,
           reason: 'should have outputted a warning');
-      expect(
-          consoleErrorMessage,
-          contains(
-              'Each child in an array or iterator should have a unique "key" prop.'));
+      expect(consoleErrorMessage,
+          contains('Each child in a list should have a unique "key" prop.'));
     });
 
+    /*
+    /// This might not be a thing in react16.
```

Comment:
Can we address this before the PR merges?

Reply:
Updated to check that it does NOT warn as this is now a valid return.